launcher/modplatform/import_ftb/PackInstallTask.cpp

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  Prism Launcher - Minecraft Launcher
+ *  Copyright (c) 2023 Trial97 <alexandru.tripon97@gmail.com>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "PackInstallTask.h"
+
+#include <QtConcurrent>
+
+#include "BaseInstance.h"
+#include "FileSystem.h"
+#include "minecraft/MinecraftInstance.h"
+#include "minecraft/PackProfile.h"
+#include "modplatform/ResourceAPI.h"
+#include "modplatform/import_ftb/PackHelpers.h"
+#include "settings/INISettingsObject.h"
+
+namespace FTBImportAPP {
+
+void PackInstallTask::executeTask()
+{
+    setStatus(tr("Copy files"));
+    setAbortable(false);
+    progress(1, 2);
+
+    m_copyFuture = QtConcurrent::run(QThreadPool::globalInstance(), [this] {
+        FS::copy folderCopy(m_pack.path, FS::PathCombine(m_stagingPath, ".minecraft"));
+        folderCopy.followSymlinks(true);
+        return folderCopy();
+    });
+    connect(&m_copyFutureWatcher, &QFutureWatcher<bool>::finished, this, &PackInstallTask::copySettings);
+    connect(&m_copyFutureWatcher, &QFutureWatcher<bool>::canceled, this, &PackInstallTask::emitAborted);
+    m_copyFutureWatcher.setFuture(m_copyFuture);
+}
+
+void PackInstallTask::copySettings()
+{
+    setStatus(tr("Copy settings"));
+    progress(2, 2);
+    QString instanceConfigPath = FS::PathCombine(m_stagingPath, "instance.cfg");
+    auto instanceSettings = std::make_shared<INISettingsObject>(instanceConfigPath);
+    instanceSettings->suspendSave();
+    instanceSettings->registerSetting("InstanceType", "OneSix");
+    instanceSettings->set("InstanceType", "OneSix");
+
+    if (m_pack.jvmArgs.isValid() && !m_pack.jvmArgs.toString().isEmpty()) {
+        instanceSettings->registerSetting("OverrideJavaArgs", true);
+        instanceSettings->set("OverrideJavaArgs", true);
+        instanceSettings->registerSetting("JvmArgs", m_pack.jvmArgs.toString());
+        instanceSettings->set("JvmArgs", m_pack.jvmArgs.toString());
+    }
+
+    MinecraftInstance instance(m_globalSettings, instanceSettings, m_stagingPath);
```

Comment:
If you set everything after constructing `MinecraftInstance`, then registration wouldn't be needed.